model/riscv_mem.sail

```diff
@@ -243,7 +206,11 @@ function mem_write_value_priv_meta (paddr, width, value, typ, priv, meta, aq, rl
   then MemException(E_SAMO_Addr_Align())
   else {
     let result = checked_mem_write(paddr, width, value, typ, priv, meta, aq, rl, con);
-    rvfi_write(paddr, width, value, meta, result);
+    if rv_enable_callbacks then
+      match result {
+        MemValue(_) => mem_update_callback(paddr, width, value, /* is_exception */ false),
+        MemException(_) => mem_update_callback(paddr, width, value, /* is_exception */ true)
```

Comment:
I don't know if it's necessarily better, but we only call out `mem_read_callback()` function on success `MemValue()`, and then we have another callback for traps in `trap_handler()`, where the `print_reg()`s are.

Maybe keep this (because you can easily ignore the exception case) but also add the trap callback.

You might want to add the exception code to the callback to make logging a bit easier:

```
      match result {
        MemValue(_) => mem_write_callback(paddr, width, value),
        MemException(e) => mem_exception_callback(paddr, width, num_of_ExceptionType(e)),
```